src/Data/Heap.hs

```diff
@@ -94,6 +94,9 @@ import Data.Bifunctor
 import qualified Data.List as L
 import Control.Applicative (Applicative(pure))
 import Control.Monad (liftM)
+#if MIN_VERSION_base(4,9,0)
```

Comment:
I'd vote to just depend on the `semigroups` library so that we don't have to use this bit of CPP.